opentracing/context.go

```diff
@@ -0,0 +1,35 @@
+package opentracing
+
+// A ContextID is the smallest amount of state needed to describe a span's
+// context within a larger [potentially distributed] trace. The ContextID is
+// not intended to encode the span's operation name, timing, or log data, but
+// merely any unique identifiers (etc) needed to contextualize it within a
+// larger trace tree.
+//
+// ContextIDs are sufficient to propagate the, well, *context* of a particular
+// trace from process to process.
+type ContextID interface {
+	// Create a child context for this ContextID, and return both that child's
+	// own ContextID as well as any Tags that should be added to the child's
+	// Span.
+	//
+	// The returned ContextID type must be the same as the type of the
+	// ContextID implementation itself.
+	NewChild() (childCtx ContextID, initialChildSpanTags Tags)
+
+	// Serializes the ContextID as an arbitrary byte string.
+	Serialize() []byte
```

Comment:
Ok, noted... I don't feel strongly about the default but I do think `[]bytes` should be an option at least. Happy to change this whenever once we have consensus about the other (IMO larger) issues.

I would be concerned with adding zipkin-style 4-number tuples here, by the way(*). Some ContextIDs won't be able to represent themselves that way, so it's problematic to add it at this layer. Thoughts?

(*): it also makes me grumpy that Zipkin sends the parent_id over the wire... should not be formally necessary to serialize along with the context, but so it goes! Minor point.